packages/task/src/browser/task-service.ts

```diff
@@ -791,10 +791,10 @@ export class TaskService implements TaskConfigurationClient {
         this.addRecentTasks(task);
         try {
             const resolver = await this.taskResolverRegistry.getTaskResolver(task.type);
-            const resolvedTask = resolver ? await resolver.resolveTask(task) : task;
+            const resolvedTask = resolver && await resolver.resolveTask(task) || task;
```

Comment:
I'm not sure using the unresolved tasks is the right thing to do here: looking at the code in VS Code, they seem to treat that condition as an error.
Think about it this way: if we have a task that is not a contributed task of type 'foo', but defined in a tasks.json file, it will  not have an execution set (and therefore no 'taskType' field). So looking up a runner using the taskType (undefined) and type ('foo') fields will fall back to using the ProcessTaskRunner, but executing a custom contributed task will fail in unforeseen ways. I think it's better to handle that condition explicitly.

Reply:
> I'm not sure using the unresolved tasks is the right thing to do here: looking at the code in VS Code, they seem to treat that condition as an error.

It looks strange for me, because:
- According to the [VS Code documentation](https://code.visualstudio.com/api/references/vscode-api#TaskProvider) (please see a description for `resolveTask()` method):
> A valid default implementation for the resolveTask method is to return undefined 

- I tested the use case for VS Code using test plugin, which uses a task resolver which returns `undefined`. The result - a task is run successfully - no errors.

Do you have steps to reproduce which I can use to confirm that behavior you described above?

Reply:
I have looked at the vscode source. I guess the test would be a task of a contributed type that is not a customization of a contributed task.

Reply:
what's the goal to
> contribute only a type, but don't return anything from 'provideTasks()'

?

Do you know a VS Code extension which has such behavior?

Reply:
No, but it's valid according to the documentation 🤷

Reply:
The goal is to make sure we hit the failure case I was describing. If you know a simpler way to reproduce the scenario, feel free to use it.

Reply:
So, if I understand correctly, the your use case is:
- a plugin contribute a type for tasks
- **don't** provide any tasks
- returns **undefined** when `resolveTask` method is called

correct?

Reply:
Yes, except the part where you're configuring a task of that type in a tasks.json file.

Reply:
I haven't tested anything so far, just inspected the code here and in VS Code.

Reply:
Relevant code starts here: https://github.com/microsoft/vscode/blob/1c01ce84b502457d61efa87c75cf5ddaf6c40f10/src/vs/workbench/api/common/extHostTask.ts